types/stylelint/index.d.ts

```diff
@@ -631,5 +631,5 @@ declare module 'stylelint' {
 
 	const stylelint: PublicApi;
 
-	export default stylelint;
+	export = stylelint;
```

Comment:
Thanks for the pull request! How about just adding a new `export`?

```suggestion
    export default stylelint;
	export = stylelint; // for CommonJS
```

Reply:
Thank you for your fast response!
After applying your suggestion, now I gotanother ts error.

```
Module '"stylelint"' can only be default-imported using the 'esModuleInterop' flagts(1259)
index.d.ts(635, 2): This module is declared with 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.
```

![image](https://user-images.githubusercontent.com/19153718/222657402-2cddcdc6-f456-440c-be76-1017958b0248.png)

Reply:
Good catch. It's difficult...

Then, does it make sense to use `require("...").default` instead?

```diff
-const stylelint = require("stylelint");
+const stylelint = require("stylelint").default;
```

Of course, we may need to change the following code:

https://github.com/stylelint/stylelint/blob/d2ca95a197669c9c18f87f7b77fa309cf4c2d5f7/lib/index.js#L35

```diff
 module.exports = stylelint; 
+module.exports.default = stylelint; 
```

Reply:
Ohh! I thought about only how to fix the type, but yeah, I think now it works fine.